scripts/train/train.py

```diff
@@ -40,29 +44,100 @@
 log = logging.getLogger(__name__)
 
 
-def validate_config(cfg: DictConfig):
+@dataclass
+class TrainConfig:
+    model: Dict[str, Any] = MISSING
+    tokenizer: Dict[str, Any] = MISSING
+    optimizer: Dict[str, Any] = MISSING
+    scheduler: Dict[str, Any] = MISSING
+    train_loader: Dict[str, Any] = MISSING
+    device_train_batch_size: int = MISSING
+    device_eval_batch_size: int = MISSING
+    max_duration: Union[int, str] = MISSING
+    eval_interval: Union[int, str] = MISSING
```

Comment:
eval_interval isn't supposed to be required

Reply:
It's in the unit test under mandatory parameters [here](https://github.com/mosaicml/llm-foundry/blob/f01f6256ba487ef9e433f49ceac68cf327ac74bb/tests/a_scripts/train/test_train_inputs.py#L67C1-L79C10), should I remove it?